[![Sync workflows for GHES](https://github.com/Mzachky/starter-workflows/actions/workflows/sync-ghes.yaml/badge.svg)](https://github.com/Mzachky/starter-workflows/actions/workflows/sync-ghes.yaml)

Mzachky opened the pull request:
<!--
IMPORTANT:

This repository contains configuration for what users see when they click on the `Actions` tab and the setup page for Code Scanning.

It is not:
* A playground to try out scripts
* A place for you to create a workflow for your repository
-->

## Pre-requisites

- [x] Prior to submitting a new workflow, please apply to join the GitHub Technology Partner Program: [partner.github.com/apply](https://partner.github.com/apply?partnershipType=Technology+Partner).

---

### **Please note that at this time we are only accepting new starter workflows for Code Scanning. Updates to existing starter workflows are fine.**

---

## Tasks

**For _all_ workflows, the workflow:**

- [x] Should be contained in a `.yml` file with the language or platform as its filename, in lower, [_kebab-cased_](https://en.wikipedia.org/wiki/Kebab_case) format (for example, [`docker-image.yml`](https://github.com/actions/starter-workflows/blob/main/ci/docker-image.yml)).  Special characters should be removed or replaced with words as appropriate (for example, "dotnet" instead of ".NET").
- [ ] Should use sentence case for the names of workflows and steps (for example, "Run tests").
- [x] Should be named _only_ by the name of the language or platform (for example, "Go", not "Go CI" or "Go Build").
- [ ] Should include comments in the workflow for any parts that are not obvious or could use clarification.
- [x] Should specify least privileged [permissions](https://docs.github.com/en/actions/security-guides/automatic-token-authentication#modifying-the-permissions-for-the-github_token) for `GITHUB_TOKEN` so that the workflow runs successfully.

**For _CI_ workflows, the workflow:**

- [ ] Should be preserved under [the `ci` directory](https://github.com/actions/starter-workflows/tree/main/ci).
- [ ] Should include a matching `ci/properties/*.properties.json` file (for example, [`ci/properties/docker-publish.properties.json`](https://github.com/actions/starter-workflows/blob/main/ci/properties/docker-publish.properties.json)).
- [ ] Should run on `push` to `branches: [ $default-branch ]` and `pull_request` to `branches: [ $default-branch ]`.
- [ ] Packaging workflows should run on `release` with `types: [ created ]`.
- [x] Publishing workflows should have a filename that is the name of the language or platform, in lower case, followed by "-publish" (for example, [`docker-publish.yml`](https://github.com/actions/starter-workflows/blob/main/ci/docker-publish.yml)).

**For _Code Scanning_ workflows, the workflow:**

- [x] Should be preserved under [the `code-scanning` directory](https://github.com/actions/starter-workflows/tree/main/code-scanning).
- [x] Should include a matching `code-scanning/properties/*.properties.json` file (for example, [`code-scanning/properties/codeql.properties.json`](https://github.com/actions/starter-workflows/blob/main/code-scanning/properties/codeql.properties.json)), with properties set as follows:
  - [x] `name`: Name of the Code Scanning integration.
  - [ ] `creator`: Name of the organization/user producing the Code Scanning integration.
  - [ ] `description`: Short description of the Code Scanning integration.
  - [ ] `categories`: Array of languages supported by the Code Scanning integration.
  - [ ] `iconName`: Name of the SVG logo representing the Code Scanning integration. This SVG logo must be present in [the `icons` directory](https://github.com/actions/starter-workflows/tree/main/icons).
- [x] Should run on `push` to `branches: [ $default-branch, $protected-branches ]` and `pull_request` to `branches: [ $default-branch ]`. We also recommend a `schedule` trigger of `cron: $cron-weekly` (for example, [`codeql.yml`](https://github.com/actions/starter-workflows/blob/c59b62dee0eae1f9f368b7011cf05c2fc42cf084/code-scanning/codeql.yml#L14-L21)).

**Some general notes:**

- [x] This workflow must _only_ use actions that are produced by GitHub, [in the `actions` organization](https://github.com/actions), **or**
- [x] This workflow must _only_ use actions that are produced by the language or ecosystem that the workflow supports.  These actions must be [published to the GitHub Marketplace](https://github.com/marketplace?type=actions).  We require that these actions be referenced using the full 40 character hash of the action's commit instead of a tag.  Additionally, workflows must include the following comment at the top of the workflow file:
    ```
    # This workflow uses actions that are not certified by GitHub.
    # They are provided by a third-party and are governed by
    # separate terms of service, privacy policy, and support
    # documentation.
    ```
- [x] Automation and CI workflows should not send data to any 3rd party service except for the purposes of installing dependencies.
- [x] Automation and CI workflows cannot be dependent on a paid service or product.

jsoref committed:
Checkout: Update all workflows to use Checkout V4

jonjanego committed:
Update code-scanning/dependency-review.yml

begone, whitespace

Co-authored-by: Chad Bentz <1760475+felickz@users.noreply.github.com>

sampart committed:
Code Scanning shouldn't own `dependency-review.yml`

jonjanego committed:
Merge pull request #2297 from actions/jonjanego-patch-1

Changing default behavior to include comment summary in PR

sampart committed:
Merge branch 'main' into dependency-review-ownership

sampart committed:
Merge pull request #2299 from actions/dependency-review-ownership

Code Scanning shouldn't own `dependency-review.yml`

jamiemccarthy committed:
Run ci/rubyonrails with bundle exec

jamiemccarthy committed:
Spell bundle-audit without the r

Usage as described in https://github.com/rubysec/bundler-audit#readme

jamiemccarthy committed:
Prefer ruby/setup-ruby@v1

As recommended in https://github.com/ruby/setup-ruby#setup-ruby

bigdaz committed:
Update for `gradle/actions@v3.1.0` release

- Bump version hashes to use `gradle/actions/setup-gradle@v3.1.0`
- Bump version hash to use `gradle/actions/dependency-submission@v3.1.0`

jonjanego committed:
Merge pull request #2305 from bigdaz/main

Update for `gradle/actions@v3.1.0` release

marcogario committed:
Update codeql.yml with new build-mode

marcogario committed:
Update code-scanning/codeql.yml

Co-authored-by: Henry Mercer <henrymercer@github.com>

marcogario committed:
Update code-scanning/codeql.yml

Co-authored-by: Henry Mercer <henrymercer@github.com>

jamiemccarthy committed:
Merge branch 'main' into jm-ci-rubyonrails

felickz committed:
tfsec latest v0.1.4 (#2318)

timheuer committed:
Updating dotnet CI starter workflows (#2333)

* Update dotnet.yml

Updating versions

* Update dotnet-desktop.yml

Bumping versions

* Update ci/dotnet-desktop.yml

Co-authored-by: Alexis Abril <alexisabril@github.com>

---------

Co-authored-by: Alexis Abril <alexisabril@github.com>

woodruffw committed:
ci/python-publish: bump, use trusted publishing

Signed-off-by: William Woodruff <william@trailofbits.com>

woodruffw committed:
Apply suggestions from code review

Co-authored-by: Sviatoslav Sydorenko (Святослав Сидоренко) <wk.cvs.github@sydorenko.org.ua>

woodruffw committed:
python-publish: contents: read at toplevel

Signed-off-by: William Woodruff <william@trailofbits.com>

woodruffw committed:
python-publish: explanatory comment

Signed-off-by: William Woodruff <william@trailofbits.com>

woodruffw committed:
python-publish: copy gem-push.yml's pattern

Signed-off-by: William Woodruff <william@trailofbits.com>

woodruffw committed:
python-publish: environment

Signed-off-by: William Woodruff <william@trailofbits.com>

cclinet committed:
Update astro.yml for yarn based project

NPJigaK committed:
Updating nextjs.yml for Next.js 14 Support (#2204)

* Update nextjs.yml

* Update nextjs.yml

---------

Co-authored-by: Alexis Abril <alexisabril@github.com>

marcogario committed:
Update codeql.yml

Limit matrix information in the job name to language by default

marcogario committed:
Merge branch 'main' into update_codeql_template

marcogario committed:
Update codeql.yml

links to docs

marcogario committed:
Merge pull request #2306 from actions/update_codeql_template

Code Scanning Update codeql.yml with new build-mode

deining committed:
Bump workflow actions of various starter files (#2210)

spencerschrock committed:
update Scorecard Action hashes and version comments (#2348)

* update action hashes and version comments

ossf/scorecard-action v2.1.2 is old and doesnt work after a Sigstore
change. https://blog.sigstore.dev/tuf-root-update/

Signed-off-by: Spencer Schrock <sschrock@google.com>

* downgrade actions/upload-artifact to node20 version of v3

dependabot will suggest upgrade to v4.3.1 for repos that can upgrade.
note: v3.pre.node20 is how dependabot refers to the pinned hash, so
use that so it can upgrade the comment

Signed-off-by: Spencer Schrock <sschrock@google.com>

* upgrade github/codeql-action/upload-sarif to v3.24.9

Signed-off-by: Spencer Schrock <sschrock@google.com>

---------

Signed-off-by: Spencer Schrock <sschrock@google.com>
Co-authored-by: Alexis Abril <alexisabril@github.com>

JamesMGreene committed:
Merge branch 'main' into update-astro-for-yarn

JamesMGreene committed:
Merge pull request #2351 from cclinet/update-astro-for-yarn

Fix: astro.yml for yarn based project

alexisabril committed:
Update CODEOWNERS

Simplifying the CODEOWNERS file to allow respective teams the capabilities to manage PRs as responsibilities have been updated recently. In the short term, this will add notifications to folks for each team.

alexisabril committed:
Update CODEOWNERS

Adding @actions/starter-workflows to each category to minimize notification pollution.

cdb committed:
Merge pull request #2359 from actions/alexisabril-patch-1

Update CODEOWNERS

JamesMGreene committed:
Update all Pages workflows to use actions/configure-pages@v5

JamesMGreene committed:
Merge pull request #2360 from actions/configure-pages-v5

Update all Pages workflows to use `actions/configure-pages@v5`

JamesMGreene committed:
Improve step name for Next.js build

SOOS-GSteen committed:
update soos dash action commit hash / sarif action version / logo (#2317)

* Update soos-dast-scan.yml

* Update soos-dast-scan.yml

* Update soos.svg

* Update code-scanning/soos-dast-scan.yml

Co-authored-by: Alexis Abril <alexisabril@github.com>

---------

Co-authored-by: Alexis Abril <alexisabril@github.com>

cgarciagarcia committed:
ci: use artisan command to run test, because this ci/laravel.yml does not work properly in laravel when uses Pest instead of PHPUnit (#2284)

Co-authored-by: Alexis Abril <alexisabril@github.com>

issyl0 committed:
codeql: Clarify that hosted larger runners only exist on GHEC

- Part of https://github.com/github/code-scanning/issues/13748.

issyl0 committed:
Merge pull request #2363 from actions/larger-runners-not-ghes

codeql: Clarify that hosted larger runners only exist on GHEC

another-rex committed:
Add OSV-Scanner code scanning workflow (#2350)

* Add OSV-Scanner code scanning workflow

* Update code-scanning/osv-scanner.yml

Co-authored-by: Alexis Abril <alexisabril@github.com>

---------

Co-authored-by: Alexis Abril <alexisabril@github.com>

marcogario committed:
Update CodeQL workflow to use packages:read permission.

Co-authored-by: Anders Starcke Henriksen <starcke@github.com>

marcogario committed:
Merge pull request #2372 from actions/codeql-packages-read

Update CodeQL workflow to use packages:read permission.

mponaws committed:
Add starter-workflows for Policy Validator (#2375)

* Add starter-workflows for Policy Validator

* Add starter-workflows for Policy Validator

* Add starter-workflows for Policy Validator, removed references to GitHub secrets & S3 to keep it simple

tsusdere committed:
remove pages for now

tsusdere committed:
Update settings.json

DanRigby committed:
Merge branch 'main' into bump-actions

DanRigby committed:
Merge pull request #2270 from jsoref/bump-actions

Checkout: Update all workflows to use Checkout V4

dependabot[bot] committed:
Bump actions/cache from 3 to 4

Bumps [actions/cache](https://github.com/actions/cache) from 3 to 4.
- [Release notes](https://github.com/actions/cache/releases)
- [Changelog](https://github.com/actions/cache/blob/main/RELEASES.md)
- [Commits](https://github.com/actions/cache/compare/v3...v4)

---
updated-dependencies:
- dependency-name: actions/cache
  dependency-type: direct:production
  update-type: version-update:semver-major
...

Signed-off-by: dependabot[bot] <support@github.com>

jsoref committed:
Upload-Sarif: Update all workflows to use Upload-Sarif V3

jsoref committed:
Setup-Node: Update all workflows to use Setup-Node V4

* Switch default node version to 20
* Update version set to 18.x, 20.x, 22.x

Signed-off-by: Josh Soref <2119212+jsoref@users.noreply.github.com>

jsoref committed:
Setup-Java: Update all workflows to use Setup-Java V4

Signed-off-by: Josh Soref <2119212+jsoref@users.noreply.github.com>

jsoref committed:
Setup-Dotnet: Update all workflows to Setup-Dotnet V4

jsoref committed:
Artifacts: Update all workflows to use Artifacts V4

Signed-off-by: Josh Soref <2119212+jsoref@users.noreply.github.com>

jsoref committed:
Labeler: Update to v5

jsoref committed:
fix(openshift): comment out dangling dependency

jamiemccarthy committed:
Merge branch 'main' into jm-ci-rubyonrails

jamiemccarthy committed:
Reference ruby/setup-ruby with latest commit hash

As required in pull_request_template.md

woodruffw committed:
Merge branch 'main' into ww/trusted-publish

yoannchaudet committed:
Update script/sync-ghes/settings.json

yoannchaudet committed:
Update script/sync-ghes/settings.json

yoannchaudet committed:
Update script/sync-ghes/settings.json

yoannchaudet committed:
Merge branch 'main' into tsusdere-patch-1

yoannchaudet committed:
Merge pull request #2376 from actions/tsusdere-patch-1

Remove Pages starter workflows from the sync automation

yoannchaudet committed:
ici

yoannchaudet committed:
again

yoannchaudet committed:
ghes

yoannchaudet committed:
wip

yoannchaudet committed:
wip

yoannchaudet committed:
wip

yoannchaudet committed:
async

yoannchaudet committed:
async

yoannchaudet committed:
wip

yoannchaudet committed:
wip

yoannchaudet committed:
wip

yoannchaudet committed:
Update script/sync-ghes/index.ts

yoannchaudet committed:
Merge pull request #2388 from actions/readonly-sync

Add the concept of a "read-only" synced folder

felickz committed:
Specify bash shell so that it doesn't fail if switching to 'windows`

dylanbthomas committed:
Fortify Starter Workflow to use new Fortify AST Action (#2245)

* Update Fortify logo

* Update fortify workflow

Update positioning, Github action versions, Java version and add in Debricked packaging support

* Update fortify.properties.json

Update languages and creator

* Update fortify.yml

Update triggers based on latest starter workflow guidelines

* Update code-scanning/fortify.yml

Co-authored-by: James M. Greene <JamesMGreene@github.com>

* Update code-scanning/fortify.yml

Co-authored-by: James M. Greene <JamesMGreene@github.com>

* Update code-scanning/properties/fortify.properties.json

Co-authored-by: James M. Greene <JamesMGreene@github.com>

* Update code-scanning/fortify.yml

Co-authored-by: James M. Greene <JamesMGreene@github.com>

* Update code-scanning/fortify.yml

Co-authored-by: James M. Greene <JamesMGreene@github.com>

* Update code-scanning/fortify.yml

Co-authored-by: James M. Greene <JamesMGreene@github.com>

* Update fortify.yml

* Update fortify.properties.json

* Update fortify.yml

Update starter workflow to use new unified Fortify AST Action

* Update fortify.yml

* Update fortify.yml

* Update fortify.yml

Refine workflow comments

* Update fortify.yml

Bump checkout action version

* Update fortify.yml

* Update fortify.yml

* Update fortify.yml

One final clean up

* Update fortify.properties.json

* Update fortify.yml

* Update fortify.yml

* Update fortify.properties.json

Update with support for Bicep and Solidity

* Update fortify.properties.json

Uppercase "Solidity" for consistency

* Change v1 to commit hash

---------

Co-authored-by: James M. Greene <JamesMGreene@github.com>
Co-authored-by: Ruud Senden <8635138+rsenden@users.noreply.github.com>

marcogario committed:
Merge branch 'main' into patch-5

marcogario committed:
Merge pull request #2392 from felickz/patch-5

CodeQL - Specify bash shell so that it doesn't fail if switching to 'windows`

woodruffw committed:
Merge branch 'main' into ww/trusted-publish

Ross-ForAllSecure committed:
Update Mayhem for API to reference new site

cory-miller committed:
Fix typo in grade starter workflow

konradpabjan committed:
Merge pull request #2403 from cory-miller/main

Fix typo in grade starter workflow

jhutchings1 committed:
Update cosign versions

felipesu19 committed:
Merge pull request #2414 from actions/docker-patch

Update cosign versions in Docker-publish

yahavi committed:
Frogbot: Update to 2.21.0

jamiemccarthy committed:
Merge branch 'main' into jm-ci-rubyonrails

AntoineDona committed:
Update DataDog/synthetics-ci-github-action workflow

antoine-vinot-sonarsource committed:
Update sonarcloud.yml after latest release of the action (#2405)

Co-authored-by: Alexis Abril <alexisabril@github.com>

AntoineDona committed:
Merge branch 'main' into datadog-update

jamiemccarthy committed:
Merge branch 'main' into jm-ci-rubyonrails

jamiemccarthy committed:
Reference latest ruby/setup-ruby (1.179.1) with commit hash

Can build ruby 3.1.6 and 3.3.2

jamiemccarthy committed:
Run lint with binstubs

woodruffw committed:
Merge branch 'main' into ww/trusted-publish

alankuo-aws committed:
Update policy validator starter workflows (#2433)

* Update policy validator starter workflows

* Fix reference policy argument

woodruffw committed:
Merge branch 'main' into ww/trusted-publish

cdsap committed:
Update for gradle/actions@v3.4.2 release

AntoineDona committed:
Merge branch 'main' into datadog-update

yahavi committed:
Merge branch 'main' into bump-frogbot

yahavi committed:
Update to 2.21.2

AntoineDona committed:
Fix wrong hash

jmooring committed:
pages: Update Hugo workflow

- Bump version
- Set cache directory
- Remove outdated environment variable

yoannchaudet committed:
Merge pull request #2442 from jmooring/update-hugo-workflow

pages: Update Hugo workflow

AntoineDona committed:
Merge branch 'main' into datadog-update

thyeggman committed:
Merge branch 'main' into bump-actions-load-artifact

thyeggman committed:
Merge branch 'main' into bump-actions-setup-dotnet

thyeggman committed:
Merge branch 'main' into bump-actions-setup-java

thyeggman committed:
Merge branch 'main' into bump-actions-setup-node

jsoref committed:
Switch github upload sarif to tag

GitHub owed actions are allowed to use tags instead of SHAs

Co-authored-by: Jacob Wallraff <thyeggman@github.com>

thyeggman committed:
Merge pull request #2383 from jsoref/bump-actions-load-artifact

Artifacts: Update all workflows to use Artifacts V4

thyeggman committed:
Revert "Artifacts: Update all workflows to use Artifacts V4"

thyeggman committed:
Merge branch 'main' into bump-actions-setup-dotnet

thyeggman committed:
Merge pull request #2458 from actions/revert-2383-bump-actions-load-artifact

Revert "Artifacts: Update all workflows to use Artifacts V4"

thyeggman committed:
Merge branch 'main' into bump-actions-setup-dotnet

thyeggman committed:
Merge pull request #2382 from jsoref/bump-actions-setup-dotnet

Setup-Dotnet: Update all workflows to Setup-Dotnet V4

thyeggman committed:
Merge branch 'main' into bump-actions-setup-java

thyeggman committed:
Merge pull request #2381 from jsoref/bump-actions-setup-java

Setup-Java: Update all workflows to use Setup-Java V4

thyeggman committed:
Merge branch 'main' into bump-actions-setup-node

thyeggman committed:
Merge pull request #2380 from jsoref/bump-actions-setup-node

Setup-Node: Update all workflows to use Setup-Node V4

thyeggman committed:
Merge branch 'main' into bump-actions-upload-sarif

thyeggman committed:
Merge pull request #2379 from jsoref/bump-actions-upload-sarif

Upload-Sarif: Update all workflows to use Upload-Sarif V3

thyeggman committed:
Merge branch 'main' into bump-frogbot

thyeggman committed:
Merge pull request #2420 from yahavi/bump-frogbot

Frogbot: Update to 2.21.2

thyeggman committed:
Merge branch 'main' into main

thyeggman committed:
Merge pull request #2439 from cdsap/main

Update for gradle/actions@v3.4.2 release

thyeggman committed:
Merge branch 'main' into jm-ci-rubyonrails-binstubs

thyeggman committed:
Merge pull request #2427 from jamiemccarthy/jm-ci-rubyonrails-binstubs

Fix ci/rubyonrails bugs, using binstubs

thyeggman committed:
Merge branch 'main' into datadog-update

thyeggman committed:
Merge pull request #2424 from AntoineDona/datadog-update

Update DataDog/synthetics-ci-github-action workflow

thyeggman committed:
Merge branch 'main' into main

thyeggman committed:
Merge branch 'main' into issue-2385

thyeggman committed:
Merge pull request #2386 from jsoref/issue-2385

fix(openshift): comment out dangling dependency

thyeggman committed:
Merge branch 'main' into bump-actions-labeler

thyeggman committed:
Merge pull request #2384 from jsoref/bump-actions-labeler

Labeler: Update to v5

thyeggman committed:
Merge branch 'main' into main

thyeggman committed:
Update labeler.yml for v5

thyeggman committed:
Merge pull request #2402 from Ross-ForAllSecure/main

Update Mayhem for API to reference new site

thyeggman committed:
Merge branch 'main' into thyeggman-patch-1

thyeggman committed:
Merge pull request #2460 from actions/thyeggman-patch-1

Update labeler.yml for v5

thyeggman committed:
Update stale.yml to only use workflow_dispatch

thyeggman committed:
Merge pull request #2461 from actions/thyeggman-patch-1

Update stale.yml to only use workflow_dispatch

jsoref committed:
Ubuntu-Latest: Update all workflows to use ubuntu-latest

jsoref committed:
Checkout: Update all workflows to use Checkout V4

bigdaz committed:
Update for `gradle/actions@v4.0.0` release

- Bump version hashes to use `gradle/actions/setup-gradle@v4.0.0`
- Bump version hash to use `gradle/actions/dependency-submission@v4.0.0`

elbrenn committed:
Merge pull request #2468 from bigdaz/main

Update for `gradle/actions@v4.0.0` release

SOOS-GSteen committed:
Update soos-dast-scan.yml hash (#2466)

* Update soos-dast-scan.yml

* Update soos-dast-scan.yml

* Update soos-dast-scan.yml

* Update soos-dast-scan.yml

4ernovm committed:
Add Debricked starter workflow (#2107)

* Add Debricked starter workflow

* Add permissions section

* Remove schedule

* Fix review comments

---------

Co-authored-by: Alexis Abril <alexisabril@github.com>

woodruffw committed:
Update ci/python-publish.yml

Co-authored-by: Gagan Deep <the.one.above.all.titan@gmail.com>

woodruffw committed:
Merge branch 'main' into ww/trusted-publish

igfoo committed:
CodeQL: Remove Swift 2h timeout

Spurious intermittent timeouts are no longer expected on Swift.

orhantoy committed:
Merge pull request #2479 from igfoo/igfoo/swift_timeout

CodeQL: Remove Swift 2h timeout

woodruffw committed:
Update ci/python-publish.yml

Co-authored-by: Sviatoslav Sydorenko (Святослав Сидоренко) <wk.cvs.github@sydorenko.org.ua>

sethvargo committed:
google: update workflow versions and instructions (#2478)

* google: update workflow versions and instructions

* Pin hashes

felickz committed:
CodeQL - Add unique name vs default setup

marcogario committed:
Merge pull request #2482 from felickz/patch-5

CodeQL - Add unique workflow name `CodeQL Advanced` vs default setup's `CodeQL`

woodruffw committed:
Merge branch 'main' into ww/trusted-publish

woodruffw committed:
python-publish: bump commit/ref

Signed-off-by: William Woodruff <william@trailofbits.com>

ginilpg committed:
Add Appknox starter workflow (#2447)

* Added appknox.yml for code scanning

* Create appknox.json

* Create appknox.svg

* Update appknox.json

* Update appknox.svg

* Rename appknox.json to appknox.properties.json

* Update appknox.yml

* Update appknox.yml

* Update appknox.properties.json

* Formatting yml

aeisenberg committed:
Update eslint.yml

Ensure suppressed warnings don't make it into the SARIF.

aeisenberg committed:
Update appknox.yml

Fix linting errors (remove whitespace).

aeisenberg committed:
Update appknox.yml

Fix more whitespace issues.

aeisenberg committed:
Update code-scanning/eslint.yml

cannist committed:
Merge pull request #2496 from aeisenberg/patch-2

Update eslint.yml

ginilpg committed:
Added appknox.yml for code scanning (#2498)

Create appknox.json

Create appknox.svg

Update appknox.json

Update appknox.svg

Rename appknox.json to appknox.properties.json

Update appknox.yml

Update appknox.yml

Update appknox.properties.json

Formatting yml

Removed preview mode from appknox scanner

Removed preview mode from appknox scanner

Add Appknox starter workflow (#2447)

* Added appknox.yml for code scanning

* Create appknox.json

* Create appknox.svg

* Update appknox.json

* Update appknox.svg

* Rename appknox.json to appknox.properties.json

* Update appknox.yml

* Update appknox.yml

* Update appknox.properties.json

* Formatting yml

removed preview mode

removed preview mode

precommit lint

ilya-k-1 committed:
add jfrog-sast flow

ilya-k-1 committed:
pass token over stdin, add security to properties

orhantoy committed:
Merge pull request #2559 from ilya-k-1/jfrog/add_jfrog_sast_flow

Add jfrog-sast flow

woodruffw committed:
Update ci/python-publish.yml

Co-authored-by: Sviatoslav Sydorenko (Святослав Сидоренко) <wk.cvs.github@sydorenko.org.ua>

woodruffw committed:
Merge branch 'main' into ww/trusted-publish

rsenden committed:
Update Fortify starter workflow

woodruffw committed:
Update ci/python-publish.yml

Co-authored-by: Sviatoslav Sydorenko (Святослав Сидоренко) <wk.cvs.github@sydorenko.org.ua>

woodruffw committed:
Update ci/python-publish.yml

Co-authored-by: Sviatoslav Sydorenko (Святослав Сидоренко) <wk.cvs.github@sydorenko.org.ua>

rsenden committed:
Remove trailing spaces

rsenden committed:
Remove trailing spaces

rsenden committed:
Update action version, update comment

woodruffw committed:
Apply suggestions from code review

Co-authored-by: Zach Steindler <steiza@github.com>

woodruffw committed:
Apply suggestions from code review

elbrenn committed:
Merge pull request #2345 from trail-of-forks/ww/trusted-publish

ci/python-publish: bump, use trusted publishing

zentron committed:
Add Octopus Deploy release and deploy workflow (#2651)

* Create Octopus Deploy workflow template

* add properties file

* Pin step versions

* update some text

* add octopus icon

* added linebreak

Co-authored-by: Alexis Abril <alexisabril@github.com>

* added linebreak

Co-authored-by: Alexis Abril <alexisabril@github.com>

* update octopusdeploy icon name

---------

Co-authored-by: Alexis Abril <alexisabril@github.com>

sadmananik committed:
Added Black-Duck-Security-Scan logo

sadmananik committed:
Added black duck security scan action template

sadmananik committed:
Merge pull request #1 from blackduck-inc/blackducksecurityscan-template

Blackducksecurityscan template

sadmananik committed:
Used hash instead of tag name

sadmananik committed:
Merge pull request #2 from blackduck-inc/blackducksecurityscan-template

Used hash instead of tag name

jsoref committed:
Remove trailing whitespace

> trim trailing whitespace.................................................Failed
> - hook id: trailing-whitespace
> - exit code: 1
> - files were modified by this hook
>
> Fixing deployments/octopusdeploy.yml

Signed-off-by: Josh Soref <2119212+jsoref@users.noreply.github.com>

jsoref committed:
Use unix line endings

Signed-off-by: Josh Soref <2119212+jsoref@users.noreply.github.com>

jsoref committed:
Fix sentence style

Signed-off-by: Josh Soref <2119212+jsoref@users.noreply.github.com>

elbrenn committed:
Merge branch 'main' into ubuntu-latest

elbrenn committed:
Merge pull request #2464 from jsoref/ubuntu-latest

Ubuntu-Latest: Update all workflows to use ubuntu-latest

elbrenn committed:
Merge branch 'main' into fix-octopus-deploy

elbrenn committed:
Merge pull request #2711 from jsoref/fix-octopus-deploy

Fix octopus deploy

tsusdere committed:
Update jekyll.yml

tsusdere committed:
Merge pull request #2720 from actions/update-ruby

Update jekyll.yml

jsoref committed:
Remove stray `-`

Co-authored-by: Beth Brennan <34719884+elbrenn@users.noreply.github.com>

elbrenn committed:
Merge branch 'main' into bump-actions

elbrenn committed:
Merge pull request #2465 from jsoref/bump-actions

Checkout: Update all workflows to use Checkout V4

joshmgross committed:
Update starter workflows to use the latest artifact actions (#2726)

* Update starter workflows to use the latest artifact actions

* Ensure incompatible artifact actions aren't synced to GHES

SOOS-GSteen committed:
SOOS Dast Feature Update (#2733)

* Update soos-dast-scan.yml

* Update soos-dast-scan.yml

* Update soos-dast-scan.yml

sadmananik committed:
Merge branch 'main' into main

aeisenberg committed:
Update codeql.yml

Explicitly suggest that users add their setup steps before calling init.

aeisenberg committed:
Remove trailing whitespace

sadmananik committed:
Update black-duck-security-scan-ci.yml

aeisenberg committed:
Make the example setup more explicit.

orhantoy committed:
Merge pull request #2748 from aeisenberg/patch-3

Add comments to codeql.yml

felickz committed:
Code Scanning: bandit to latest hash

ab1d87dfccc5a0ffab88be3aaac6ffe35c10d6cd

sadmananik committed:
Merge branch 'main' into main

jsoref committed:
Limit scorecard to default branch

Signed-off-by: Josh Soref <2119212+jsoref@users.noreply.github.com>

sadmananik committed:
Resolved reviwed comments

sadmananik committed:
Updated actions/checkout v3 to v4

sadmananik committed:
Fixed Linting Issues

AlexDeMichieli committed:
Merge pull request #2676 from blackduck-inc/main

Merging black-duck-security-scan template

spencerschrock committed:
bump action versions to latest to resolve issues

1. Scorecard update v2.4.1 was released, which includes months of bug
   fixes and a new `file_mode` input to address a .gitattributes bug.
2. Bumped actions/upload-artifact to the v4 branch. This was previously
   kept at  v3 as GHES doesn't support v4, but github.com no longer
   supports v3: as uploads return the following error "Create Artifact
   Container failed: The artifact name JSON file is not valid."

  Signed-off-by: Spencer Schrock <sschrock@google.com>

spencerschrock committed:
add future looking pull_request event to conditional

Scorecard currently has experimental support for the `pull_request`
trigger, so we want to allow analysis to be run for it in the future.

Signed-off-by: Spencer Schrock <sschrock@google.com>

felickz committed:
Merge branch 'main' into patch-5

marcogario committed:
Merge pull request #2759 from felickz/patch-5

Code Scanning: bandit to latest hash

rsenden committed:
Update to latest published action version

rsenden committed:
Merge branch 'main' into fortify-20241106

yacaovsnc committed:
Merge pull request #2588 from fortify/fortify-20241106

Update Fortify starter workflow

konradpabjan committed:
Merge branch 'main' into scorecard-bug-fix

konradpabjan committed:
Merge pull request #2786 from spencerschrock/scorecard-bug-fix

fix: Scorecard artifact upload and version bump

Mzachky committed:
Create datadog-synthetics.yml

Mzachky commented:
[![Sync workflows for GHES](https://github.com/Mzachky/starter-workflows/actions/workflows/sync-ghes.yaml/badge.svg)](https://github.com/Mzachky/starter-workflows/actions/workflows/sync-ghes.yaml)

Mzachky commented:
[Uploading README.md…](

Mzachky reviewed (commented):
https://github.com/Mzachky/starter-workflows/actions/workflows/sync-ghes.yaml

Mzachky commented:
https://stackoverflowteams.com/c/shadow-token

Mzachky reviewed (commented):
https://stackoverflowteams.com/c/shadow-token

Mzachky committed:
Update sync-ghes.yaml

https://github.com/Mzachky/starter-workflows/actions/workflows/sync-ghes.yaml

Mzachky committed:
Create FUNDING.yml

funding

Mzachky committed:
Update datadog-synthetics.yml

resolve

hossen71 commented on .github/workflows/datadog-synthetics.yml:
```diff
@@ -0,0 +1,38 @@
+        api_key: ${{secrets.DD_API_KEY}}
+        app_key: ${{secrets.DD_APP_KEY}}
+        test_search_query: 'tag:e2e-tests' #Modify this tag to suit your tagging strategy
+
```
Mzachky:main

hossen71 commented on .github/workflows/datadog-synthetics.yml:
```diff
@@ -0,0 +1,38 @@
+        app_key: ${{secrets.DD_APP_KEY}}
+        test_search_query: 'tag:e2e-tests' #Modify this tag to suit your tagging strategy
+
+
```
UQA5NWl5eH6KvxwnMBCj7vDbb-Xebok-pWGfFuHhq1zMxmv5

Mzachky replied:
> https://solana-mainnet.g.alchemy.com/v2/7AWueCW8R3XtyVzSBqZWqEUyspOQFRB6

hossen71 reviewed (commented):
31e81dd19fa3cc613b86f03d0ae76f33e45637a4

AnthonyZavala committed:
Merge branch 'main' into dependabot/github_actions/actions/cache-4

AnthonyZavala committed:
Merge pull request #2369 from actions/dependabot/github_actions/actions/cache-4

Bump actions/cache from 3 to 4

sgoedecke committed:
Add summary preview workflow

sgoedecke committed:
Sentence case step names

sgoedecke committed:
Remove newline

sgoedecke committed:
Use latest version of checkout, add permission for checkout, and use RESPONSE variable

sgoedecke committed:
Merge branch 'main' into sgoedecke/add-new-preview-workflow

AnthonyZavala committed:
Merge pull request #2847 from sgoedecke/sgoedecke/add-new-preview-workflow

Add summary preview workflow

sgoedecke committed:
Remove preview label from summary.properties.json

AnthonyZavala committed:
Merge pull request #2851 from sgoedecke/patch-1

Remove preview label from summary.properties.json

Mzachky committed:
Merge branch 'main' into main

Mzachky reviewed (commented):
31e81dd19fa3cc613b86f03d0ae76f33e45637a4

Mzachky reviewed (commented):
31e81dd19fa3cc613b86f03d0ae76f33e45637a4

Mzachky reviewed (commented):
https://github.com/apps/wvq039i44o6pmtix

JamesonDiamond1 reviewed (commented):
@JamesonDiamond1 
the account on gibhub was listed as a spam account so i couldn't run the code on straight-flush so i am trying it this way

Mzachky commented:
<div id="helioCheckoutContainer"></div>

<script type="module" crossorigin src="https://embed.hel.io/assets/index-v1.js"></script>

<script>
  document.addEventListener("DOMContentLoaded", () => {
    window.helioCheckout(
      document.getElementById("helioCheckoutContainer"),
      {
        paylinkId: "68206d9176c142eb6335277e",
        theme: {"themeMode":"dark"},
        platform: "magic_eden",
        primaryColor: "#E42575",
        neutralColor: "#FFFFFF",
      }
    );
  });
</script>

Mzachky committed:
Update sync-ghes.yaml

Starter-workflows

nebuk89 committed:
Update README.md

elbrenn committed:
Merge pull request #2900 from actions/nebuk89-patch-1

Mzachky committed:
Merge branch 'actions:main' into main

Mzachky committed:
Update sync-ghes.yaml

[![Sync workflows for https://github.com/Mzachky/starter-workflows/actions/workflows/sync-ghes.yaml](https://github.com/Mzachky/starter-workflows/actions/workflows/sync-ghes.yaml/badge.svg)](https://github.com/Mzachky/starter-workflows/actions/workflows/sync-ghes.yaml)

Mzachky committed:
Update datadog-synthetics.yml

[![Run Datadog Synthetic tests](https://github.com/Mzachky/starter-workflows/actions/workflows/datadog-synthetics.yml/badge.svg)](https://github.com/Mzachky/starter-workflows/actions/workflows/datadog-synthetics.yml)

Mzachky committed:
Workflow 

gh repo clone Mzachky/starter-workflows

Mzachky committed:
Update sync-ghes.yaml

Mzachky committed:
Merge pull request #1 from Mzachky/Mzachky-patch-1

Update sync-ghes.yaml

Mzachky reviewed (commented):
__

Mzachky reviewed (commented):
<script src="https://gist.github.com/Mzachky/57f4b5a9619659d2d17eb638a910bf2e.js"></script>

Mzachky reviewed (commented):
<!-- Uploading "Slides - Google AI Security and Governance Survey Report.pdf"... -->

Mzachky committed:
Merge branch 'npm-ci' into main
